_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -3253,6 +3259,32 @@ mfxStatus ImplementationAvc::QueryStatus(
     return MFX_ERR_NONE;
 }
 
+void ImplementationAvc::InitDTSList()
+{
+    mfxExtCodingOption2 * extOpt2 = GetExtBuffer(m_video);
+
+    // implemented the buffering through padding unknown DTS in the list head.
+    m_decodeTimeStamp.clear();
+    if (m_video.mfx.GopRefDist > 1)
```

Comment:
I suggest to call function "MfxHwH264Encode::GetNumReorderFrames(..)" to calculate "padding" size (to not duplicate implementation of same logic in several places in the code).